### PR TITLE
Improve error messages when unable to fetch or marshal nais.yaml

### PR DESCRIPTION
**Pull request by giacung**
* Log if unable to fetch a given manifest url.
* Refactor logic for using default manifest urls.

### CODE REVIEW

File: api/manifest.go [hunk 1]
```diff
@@ -132,27 +132,24 @@ func GenerateManifest(deploymentRequest NaisDeploymentRequest) (naisManifest Nai
...
-	urls := createManifestUrl(deploymentRequest.Application, deploymentRequest.Version)
 	for _, url := range urls {
 		manifest, err := fetchManifest(url)
 		if err == nil {
 			return manifest, nil
 		}
 	}
 
-	return NaisManifest{}, fmt.Errorf("No manifest found on the URLs %s, or the url %s\n", urls, manifestUrl)
+	return NaisManifest{}, fmt.Errorf("Error fetching or parsing manifests from URL(s):  %s", urls)
```
[the remark this answers is not in the document]
Comment (gtcno) [reply api/manifest.go]: Its allready  logged in fetchmanifest

[the remark this answers is not in the document]
Comment (Kyrremann) [reply api/manifest.go]: But it's not logged to the user. I think that is what @davidsteinsland means.

[the remark this answers is not in the document]
Comment (gtcno) [reply api/manifest.go]: True. Agreed

[the remark this answers is not in the document]
Comment (giacung) [reply api/manifest.go]: fixed-

**Pull request merged**